Web tests

Senerro opened the pull request:
Added 5 tests: 1 (login-logout) for catalog.onliner and 4 for 21vek.by

Senerro committed:
deleted api files from not uproved branch

Senerro committed:
Merge branch 'main' into AutoTesting

Senerro committed:
added prototype of amazon webtest

Senerro committed:
created test that cheks log buttons

Senerro committed:
added service for providing users

Senerro committed:
added /_testdata.properties tom git.ignor

Senerro committed:
added once more

Senerro committed:
added test of verifying selected product

Senerro committed:
added test comporated products in cards with selected products

Senerro committed:
added totalcost comparison on the CartPage

Senerro committed:
added tets that verify product  deletion  from cart

Senerro committed:
purged spaces and unusage items

Senerro committed:
fixed some things

Senerro committed:
Merge branch 'main' into WEB-tests

# Conflicts:
#	DemoAutostesting/.gitignore

aostapenko92 commented on DemoAutostesting/src/test/java/com/solvd/demoautotesting/web/catalog_onliner/HomePageTest.java:
```diff
@@ -0,0 +1,96 @@
+        loginService.login(validMyselfUser, page.clickToLoginButton());
+
+        page.clickOnProfileImage();
+        Assert.assertTrue(page.getHeader().getUserBarComponent().getProfileComponent().getLogoutButton().isElementPresent(1), "logout button isn't presented, login wasn't completed");
```
Мы не работает с элементами в тесте, getLogoutButton().isElementPresent(1) 
Лучше было бы получать Header в тесте при вызове page.clickOnProfileImage(), а далее рабоатать с методами Header, в том числе нужно создать в Header ,ekbfy метод isLogoutButtonPresent.

aostapenko92 commented on DemoAutostesting/src/main/java/com/solvd/demoautotesting/web/catalog_onliner/pages/MainLoginPage.java:
```diff
@@ -0,0 +1,43 @@
+        submitButton.click();
+    }
+
+    public ExtendedWebElement getLoginInput() {
```
Геттеры и сеттеры мы не используем на страницах, работаем с элементами в пределах класса

aostapenko92 commented on DemoAutostesting/src/test/java/com/solvd/demoautotesting/web/catalog_onliner/HomePageTest.java:
```diff
@@ -0,0 +1,96 @@
+        HomePage page = getHomePage();
+        MainLoginPage loginPage = page.clickToLoginButton();
+
+        Assert.assertTrue(loginPage.getLoginInput().isElementPresent(1), "login input isn't present");
```
нужен булиан метод isLoginInputPresent(), аналогично и на следующих строках

aostapenko92 commented on DemoAutostesting/src/test/java/com/solvd/demoautotesting/web/vek21/HomePageTest.java:
```diff
@@ -0,0 +1,90 @@
+    @Test
+    public void verifyScrolling() {
+        HomePage homePage = getHomePage();
+        Assert.assertTrue(homePage.getSaleProductsComponent().getProductNameByIndex(2).isElementNotPresent(2));
```
Тут аналогично, нужен ,булиан метод isProductByIndexPrecent(int index)

Senerro committed:
fixed remarks

Senerro committed:
added prototype of verify favorites button test

Senerro committed:
added verifyFavoritesButton without click on favoritesButton

this button isn't clickable and right now I unfortunately don't know why it happens

Senerro committed:
added some fixs
